operations/experimental/migrate-cf-mysql-to-pxc.yml

```diff
@@ -159,3 +167,63 @@
     options:
       ca: pxc_server_ca
       common_name: sql-db.service.cf.internal
+
+- type: replace
+  path: /instance_groups/name=api/jobs/name=cloud_controller_ng/properties/ccdb/address?
+  value: sql-db.service.cf.internal
+
+- type: replace
+  path: /instance_groups/name=cc-worker/jobs/name=cloud_controller_worker/properties/ccdb/address?
+  value: sql-db.service.cf.internal
+
+- type: replace
+  path: /instance_groups/name=scheduler/jobs/name=cloud_controller_clock/properties/ccdb/address?
+  value: sql-db.service.cf.internal
+
+- type: replace
+  path: /instance_groups/name=uaa/jobs/name=uaa/properties/uaadb/address?
+  value: sql-db.service.cf.internal
+
+- type: replace
+  path: /instance_groups/name=api/jobs/name=routing-api/properties/routing_api/sqldb/ca_cert?
+  value: "((pxc_server_ca.certificate))"
+
+- type: replace
+  path: /instance_groups/name=api/jobs/name=cloud_controller_ng/properties/ccdb/ca_cert?
+  value: "((pxc_server_ca.certificate))"
+
+- type: replace
+  path: /instance_groups/name=cc-worker/jobs/name=cloud_controller_worker/properties/ccdb/ca_cert?
+  value: "((pxc_server_ca.certificate))"
+
+- type: replace
+  path: /instance_groups/name=scheduler/jobs/name=cloud_controller_clock/properties/ccdb/ca_cert?
+  value: "((pxc_server_ca.certificate))"
+
+- type: replace
+  path: /instance_groups/name=diego-api/jobs/name=bbs/properties/diego/bbs/sql/ca_cert?
+  value: "((pxc_server_ca.certificate))"
+
+- type: replace
+  path: /instance_groups/name=diego-api/jobs/name=bbs/properties/diego/bbs/sql/require_ssl?
+  value: true
+
+- type: replace
+  path: /instance_groups/name=diego-api/jobs/name=locket/properties/diego/locket/sql/ca_cert?
+  value: "((pxc_server_ca.certificate))"
+
+- type: replace
+  path: /instance_groups/name=diego-api/jobs/name=locket/properties/diego/locket/sql/require_ssl?
+  value: true
+
+- type: replace
+  path: /instance_groups/name=uaa/jobs/-
+  value:
+    name: ca_certs
```

Comment:
Similarly, I wonder if this should also have a guard in place:
```
- type: replace
  path: /instance_groups/name=uaa/jobs/name=ca_certs?
  value:
  ...
```

Or even:
```
- type: replace
  path: /instance_groups/name=uaa/jobs/name=ca_certs?
  value:
  ...
```

Reply:
@jpalermo and I think the current implementation (below) is actually safer. The os-conf job does not take a list of certs, it requires them to all be concatenated into a single string. If the operator gets two ca_certs jobs, the deploy should fail fast, and hopefully they can figure out how to add an ops file with all the certs they need to provide to UAA concatenated.
```
- type: replace
  path: /instance_groups/name=uaa/jobs/-
  value:
    name: ca_certs
    release: os-conf
    properties:
      certs: "((pxc_server_ca.certificate))"
```

If we instead add the guard (`path: /instance_groups/name=uaa/jobs/name=ca_certs?`), then if the operator has two copies of this job, it won't fail until after deployment when something (either the DB or the job using the other cert) fails to talk to UAA. That sounds like a more painful debugging experience.

We opened an issue for the os-conf release about fixing this in the future, but it seems like the mediocre but best option for now. https://github.com/cloudfoundry/os-conf-release/issues/34

Reply:
Heh, I realize now that my second implementation suggestion is identical to the first. I think I meant this:
```
- type: replace
  path: /instance_groups/name=uaa/jobs/name=ca_certs?
  value:
    name: ca_certs
    release: os-conf

- type: replace
  path: /instance_groups/name=uaa/jobs/name=ca_certs/properties
  value:
    certs: ...
```

However, in light of your comment that the certs can't be appended, it seems reasonable to leave it as you originally implemented it.